drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/models/ScenarioGridModel.java

```diff
@@ -100,21 +93,21 @@ public void appendNewRow(GridRow row) {
      * This method <i>insert</i> a row to the grid and populate it with values taken from given <code>Scenario</code>
      * @param row
      */
-    public void insertRow(final int rowIndex,
-                          final GridRow row, final Scenario scenario) {
+    public void insertRowGridOnly(final int rowIndex,
+                                  final GridRow row, final Scenario scenario) {
         checkSimulation();
-        insertRow(rowIndex, row);
+        super.insertRow(rowIndex, row);
         scenario.getUnmodifiableFactMappingValues().forEach(value -> {
             FactIdentifier factIdentifier = value.getFactIdentifier();
             ExpressionIdentifier expressionIdentifier = value.getExpressionIdentifier();
             if (value.getRawValue() instanceof String) {
                 String stringValue = (String) value.getRawValue();
                 int columnIndex = simulation.getSimulationDescriptor().getIndexByIdentifier(factIdentifier, expressionIdentifier);
-                setCell(rowIndex, columnIndex, () -> new ScenarioGridCell(new ScenarioGridCellValue(stringValue)));
+                setCellGridOnly(rowIndex, columnIndex, () -> new ScenarioGridCell(new ScenarioGridCellValue(stringValue)));
             } else if (value.getRawValue() instanceof Integer) {
                 String stringValue = ((Integer) value.getRawValue()).toString();
                 int columnIndex = simulation.getSimulationDescriptor().getIndexByIdentifier(factIdentifier, expressionIdentifier);
-                setCell(rowIndex, columnIndex, () -> new ScenarioGridCell(new ScenarioGridCellValue(stringValue)));
+                setCellGridOnly(rowIndex, columnIndex, () -> new ScenarioGridCell(new ScenarioGridCellValue(stringValue)));
             } else {
                 throw new UnsupportedOperationException("Only string is supported at the moment");
             }
```

Comment:
I had look on this method and think we could simplify slightly, let me know what do you think:
```java
    public void insertRowGridOnly(final int rowIndex,
                                  final GridRow row, final Scenario scenario) {
        checkSimulation();
        super.insertRow(rowIndex, row);
        scenario.getUnmodifiableFactMappingValues().forEach(value -> {
            FactIdentifier factIdentifier = value.getFactIdentifier();
            ExpressionIdentifier expressionIdentifier = value.getExpressionIdentifier();
            if (value.getRawValue() instanceof String || value.getRawValue() instanceof Integer) {
                String stringValue = String.valueOf(value.getRawValue());
                int columnIndex = simulation.getSimulationDescriptor().getIndexByIdentifier(factIdentifier, expressionIdentifier);
                setCellGridOnly(rowIndex, columnIndex, () -> new ScenarioGridCell(new ScenarioGridCellValue(stringValue)));
            } else {
                throw new UnsupportedOperationException("Only string and integer are supported at the moment");
            }
        });
        updateIndexColumn();
    }
```

Reply:
I would prefer keep things separated - I'm not sure we will actually manage integer columns, in which case I would just delete the "else"

Reply:
ok